test/e2e/specs/delete-prohibited/recordset/four-facet-selections.spec.ts

```diff
@@ -0,0 +1,95 @@
+
+// utils
+import { getCatalogID } from '@isrd-isi-edu/chaise/test/e2e/utils/catalog-utils';
+import { 
+  openFacet, openRecordsetAndResetFacetState, testSelectFacetOption
+} from '@isrd-isi-edu/chaise/test/e2e/utils/recordset-utils';
+
+const testParams: any = {
+  schema_name: 'faceting',
+  table_name: 'main',
+  sort: '@sort(id)',
+  totalNumFacets: 23,
+  defaults: {
+    numRows: 1,
+    pageSize: 25
+  },
+  multipleFacets: [
+    // F1
+    { facetIdx: 10, option: 2, numOptionsLoad: 11, numOptionsCumulative: 11, numRows: 10 },
+    // to_name
+    { facetIdx: 11, option: 0, numOptionsLoad: 10, numOptionsCumulative: 2, numRows: 5 },
+    // f3 (term)
+    { facetIdx: 12, option: 1, numOptionsLoad: 3, numOptionsCumulative: 2, numRows: 5 },
+    // from_name
+    { facetIdx: 13, option: 2, numOptionsLoad: 11, numOptionsCumulative: 6, numRows: 1 }
+  ]
+}
+
+test('Testing four facet selections 1 at a time,', async ({ page, baseURL }, testInfo) => {
+  await openRecordsetAndResetFacetState(page,
+    `${baseURL}/recordset/#${getCatalogID(testInfo.project.name)}/${testParams.schema_name}:${testParams.table_name}${testParams.sort}`,
+    testParams.totalNumFacets,
+    [0, 1, 11],
+    testParams.defaults.pageSize
+  );
+
+  await test.step('selecting facet options and verifying row after each selection', async () => {
+    for await (const [index, facetParams] of testParams.multipleFacets.entries()) {
+      // eslint-disable-next-line max-len
+      await test.step(`for facet at index: ${facetParams.facetIdx}, it should open the facet, select a value to filter on, and update the search criteria.`, async () => {
+        const facet = RecordsetLocators.getFacetById(page, facetParams.facetIdx);
+
+        // numOptionsCumulative meaning the number of options is constrained by each previous facet selection
+        await openFacet(page, facet, facetParams.facetIdx, facetParams.numOptionsCumulative, 1);
+        await expect.soft(RecordsetLocators.getCheckedFacetOptions(facet)).toHaveCount(0);
+        await testSelectFacetOption(page, facet, facetParams.option, facetParams.numRows, index + 1);
+        
+        // close the facet
+        await RecordsetLocators.getFacetHeaderButtonById(facet, facetParams.facetIdx).click();
+      });
+    };
+  });
+});
+
+test('Testing four facet selections in quick sequence and verifying data after all selections', async ({ page, baseURL }, testInfo) => {
+
+  await openRecordsetAndResetFacetState(page,
+    `${baseURL}/recordset/#${getCatalogID(testInfo.project.name)}/${testParams.schema_name}:${testParams.table_name}${testParams.sort}`,
+    testParams.totalNumFacets,
+    [0, 1, 11],
+    testParams.defaults.pageSize
+  );
+
+  await test.step('should open facets, click an option in each, and verify the data after', async () => {
+    const numFacets = testParams.multipleFacets.length;
+
+    const facet1 = RecordsetLocators.getFacetById(page, testParams.multipleFacets[0].facetIdx);
```

Comment:
Is there a reason why you're not using `for` loop here? I don't think you need to rewrite this and I'm just curious.

Reply:
I think it was this way in protractor because they were opened in the reverse order so the facet headers didn't "move" as each was opened. Then we tested them in the same order as the previous test.

Since it is reverse order for opening and then normal order for testing, I guess I didn't want to have 2 for loops? But that shouldn't matter anymore and I can change it.